deps/build.jl

```diff
@@ -1,68 +1,116 @@
+]
+
+# Download and install binaries
+for dependency in dependencies
+    download(dependency, basename(dependency))
+    evalfile(basename(dependency))
+end
+deps_prefix = joinpath(@__DIR__,"usr")
+
+# remember the working directory so that we can restore it later on
+orig_dir = pwd()
+
+println("Compiling GAP ...")
+rm(gap_bin_root, force=true, recursive=true)
+mkpath(gap_bin_root)
+cd(gap_bin_root) do
+    # initiate an out of tree build; the ARCHEXT ensures we depend on the Julia version,
+    # so that e.g. Julia 1.3 and 1.4 get separate binaries
+    run(`$(gap_src_root)/configure
+                --with-gc=julia
+                --with-julia=$(Sys.BINDIR)
+                --with-gmp=$(deps_prefix)
+                --with-readline=$(deps_prefix)
+                --with-zlib=$(deps_prefix)
+                LIBS="-Wl,-rpath -Wl,$(deps_prefix)/lib"
+                ARCHEXT=v$(VERSION)
+                `)
+    run(`make -j$(Sys.CPU_THREADS)`)
+end
+
+# clean out some clutter
+rm(joinpath(gap_bin_root, "obj"), force=true, recursive=true)
 
 gap_build_packages =  get(ENV, "GAP_BUILD_PACKAGES", "no")
 if gap_build_packages == "yes"
-    cd("pkg")
-    # eliminate a few big packages that take long to compile
-    pkgs = Base.Filesystem.readdir()
-    pkgs = Base.filter(x -> occursin(r"^(Normaliz|semigroups|simpcomp)", x), pkgs)
-    run(`rm -rf $pkgs`)
-    run(`../bin/BuildPackages.sh`)
+    cd("$(gap_src_root)/pkg") do
+        # eliminate a few big packages that take long to compile
+        pkgs = Base.Filesystem.readdir()
+        pkgs = Base.filter(x -> occursin(r"^(Normaliz|semigroups|simpcomp)", x), pkgs)
+        run(`rm -rf $pkgs`)
+        run(`../bin/BuildPackages.sh --with-gaproot=$(gap_bin_root)`)
+    end
 elseif gap_build_packages == "debug"
-    cd("pkg")
-    pkgs = Base.Filesystem.readdir()
-    pkgs = Base.filter(x -> occursin(r"^(io|profiling)", x), pkgs)
-    run(`../bin/BuildPackages.sh $pkgs`)
+    cd("$(gap_src_root)/pkg") do
+        pkgs = Base.Filesystem.readdir()
+        pkgs = Base.filter(x -> occursin(r"^(io|profiling)", x), pkgs)
+        run(`../bin/BuildPackages.sh --with-gaproot=$(gap_bin_root) $pkgs`)
+    end
 end
 
 ##
 ## Compile JuliaInterface
 ##
 println("Compiling JuliaInterface ...")
-cd(abspath(joinpath(@__DIR__, "..", "pkg", "JuliaInterface")))
-run(`./configure $gap_root`)
-run(`make -j$(Sys.CPU_THREADS)`)
+cd(joinpath(extra_gap_root, "pkg", "JuliaInterface")) do
+    run(`./configure $gap_bin_root`)
+    run(`make -j$(Sys.CPU_THREADS)`)
+end
 
 ##
 ## Write deps.jl file containing the necessary paths
 ##
-println("Generating deps.jl ...")
+println("Generating $(deps_jl) ...")
 
 deps_string = """
-GAPROOT = "$gap_root"
+GAPROOT = "$gap_bin_root"
 """
 
-path = abspath(joinpath(@__DIR__, "deps.jl"))
-println(path)
-open(path, "w") do outputfile
-    print(outputfile,deps_string)
+println(deps_jl)
+open(deps_jl, "w") do outputfile
```

Comment:
I just learned about `write(filename, x)` which does exactly the same for `x::String`; You may, or may not want to it ;-)

Reply:
Done!